### PR TITLE
i#2626 AArch64 encoder: Simplify fsz/fsz16 operands.

### PR DESCRIPTION
**Pull request by fhahn**
Currently both the fsz16 and fsz operands were looking at bits 21 and
22, while they only need to look at bit 22. This fix allows us to use
fsz/fsz16 for other encoding classes, like the ones used for indexed FP
instructions.

It also moves the operands to their right position.

Issue #2626

### CODE REVIEW

File: core/arch/aarch64/codec.c [hunk 1]
```diff
@@ -2727,18 +2726,15 @@ encode_opnd_fsz(uint enc, int opcode, byte *pc, opnd_t opnd, OUT uint *enc_out)
 static inline bool
 decode_opnd_fsz16(uint enc, int opcode, byte *pc, OUT opnd_t *opnd)
 {
-    if (((enc >> 21) & 0x03) == 0x02) {
-        *opnd = opnd_create_immed_int(FSZ_HALF, OPSZ_2b);
-        return true;
-    }
-    return false;
+
```
Comment (egrimley) [inline core/arch/aarch64/codec.c]: Are you adding a blank line here?

File: core/arch/aarch64/codec.c [hunk 2]
```diff
@@ -2727,18 +2726,15 @@ encode_opnd_fsz(uint enc, int opcode, byte *pc, opnd_t opnd, OUT uint *enc_out)
...
+    return true;
 }
 
 static inline bool
 encode_opnd_fsz16(uint enc, int opcode, byte *pc, opnd_t opnd, OUT uint *enc_out)
 {
     if (opnd_get_immed_int(opnd) == FSZ_HALF) {
-        *enc_out = 0x02 << 21;
         return true;
     }
```
Comment (egrimley) [inline core/arch/aarch64/codec.c]: Braces are now redundant.

File: core/arch/aarch64/codec.c [hunk 3]
```diff
@@ -44,7 +44,6 @@
 #include "disassemble.h"
 #include "instr.h"
 #include "instr_create.h"
-
```
Comment (egrimley) [inline core/arch/aarch64/codec.c]: Or maybe put into alphabetical order with the other #includes.

File: core/arch/aarch64/codec.txt [hunk 4]
```diff
@@ -55,6 +55,7 @@
 
 --------------------------------  impx30     # implicit X30 operand
 --------------------------------  lsl        # implicit LSL for ADD/MOV (immediate)
+--------------------------------  fsz16      # element width of FP vector reg (half (0x2) encoding)
```
Comment (egrimley) [inline core/arch/aarch64/codec.txt]: Is the comment still correct?

File: core/arch/aarch64/codec.txt [hunk 5]
```diff
@@ -123,6 +124,7 @@
 ----------xxxxxxxxxxxxxxxxx-----  mem12q     # size is 16 bytes
 ----------xxxxxxxxxxxxxxxxx-----  prf12      # size is 0 bytes (prefetch variant of mem12)
 ---------x----------------------  imm12sh    # shift for ADD/SUB (immediate); '0x'
+---------x----------------------  fsz        # element width of FP vector reg (single (0x1) and double (0x3) encoding)
```
Comment (egrimley) [inline core/arch/aarch64/codec.txt]: Is the comment still correct? Could it be shorted to no more than 90 chars? (I know the 90-char rule is necessarily broken elsewhere in this file.)

Comment (fhahn) [reply core/arch/aarch64/codec.txt]: Updated, I've split it up across 2 lines, as I think it is helpful to have more details here.

File: core/arch/aarch64/codec.txt (hunk 4 above)
Comment (fhahn) [reply core/arch/aarch64/codec.txt]: Updated the comment to make it clear this is mostly used to distinguish between FP16 and single and double encodings in a symmetric fashion

File: core/arch/aarch64/codec.c (hunk 2 above)
Comment (fhahn) [reply core/arch/aarch64/codec.c]: Done

File: core/arch/aarch64/codec.c (hunk 1 above)
Comment (fhahn) [reply core/arch/aarch64/codec.c]: Done

**Pull request merged**